1089 how expose prom route (#152)
infrawatch/documentation: doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc

// Module included in the following assemblies:
//
// <List assemblies here, each on a new line>

// This module can be included from assemblies using the following include statement:
// include::<path>/proc_exposing-routes.adoc[leveloffset=+1]

// The file name and the ID are based on the module title. For example:
// * file name: proc_doing-procedure-a.adoc
// * ID: [id='proc_doing-procedure-a_{context}']
// * Title: = Doing procedure A
//
// The ID is used as an anchor for linking to the module. Avoid changing
// it after the module has been published to ensure existing links are not
// broken.
//
// The `context` attribute enables module reuse. Every module's ID includes
// {context}, which ensures that the module has a unique ID even if it is
// reused multiple times in a guide.
//
// Start the title with a verb, such as Creating or Create. See also
// _Wording of headings_ in _The IBM Style Guide_.
[id="exposing-routes_{context}"]
= Creating a route in {OpenShift}

> In {OpenShift} ingress traffic for applications is done by exposed services and accessing the application via a route. For more information about Configuring ingress cluster traffic, see https://docs.openshift.com/container-platform/{SupportedOpenShiftVersion}/networking/configuring_ingress_cluster_traffic/overview-traffic.html[Configuring ingress cluster traffic].

Reviewer: I find the first sentence hard to parse. How about something a bit simpler, like "In {OpenShift}, applications can be exposed to the external network via a route."

Author: Will review tomorrow thanks for picking this out. As noted earlier I didn't do my normal 2nd and 3rd readings and just brain dumped.

Author: Ah sweet Joanne already fixed this :)

> In {OpenShift} ingress traffic for applications is done by exposed services and accessing the application via a route. For more information about Configuring ingress cluster traffic, see https://docs.openshift.com/container-platform/{SupportedOpenShiftVersion}/networking/configuring_ingress_cluster_traffic/overview-traffic.html[Configuring ingress cluster traffic].

Reviewer 2: ```suggestion
In {OpenShift}, you can expose applications to the external network via a route. For more information, see https://docs.openshift.com/container-platform/{SupportedOpenShiftVersion}/networking/configuring_ingress_cluster_traffic/overview-traffic.html[Configuring ingress cluster traffic].
```

I agree with @csibbitt and I like his suggested edit.

Author: Thanks! I wrote this really fast and did not review like I would normally do and instead relied on our awesome team to fix it :)

> By default in {Project} routes are not exposed in order to limit the attack surface of {ProjectShort} deployments. In order to access some services deployed in {ProjectShort} the services in {OpenShiftShort} must be exposed for access.

Reviewer 2: ```suggestion
Routes are not exposed by default in {Project} to limit the attack surface of {ProjectShort} deployments. To access some services deployed in {ProjectShort}, you must expose the services in {OpenShiftShort} for access.
```

> The way services are exposed can be environment specific, but a common service to expose in {ProjectShort} is Prometheus.

Reviewer 2: ```suggestion
A common service to expose in {ProjectShort} is Prometheus, as shown in the following example:
```

.Procedure

. Log in to {OpenShift}.
. Change to the `service-telemetry` namespace:
+
[source,bash]
----
$ oc project service-telemetry
----

> . List the available services in the `service-telemetry` project

Reviewer 2: ```suggestion
. List the available services in the `service-telemetry` project:
```

alertmanager-operated                    ClusterIP   None             <none>        9093/TCP,9094/TCP,9094/UDP                      93m
default-cloud1-ceil-meter-smartgateway   ClusterIP   172.30.114.195   <none>        8081/TCP                                        93m
default-cloud1-coll-meter-smartgateway   ClusterIP   172.30.133.180   <none>        8081/TCP                                        93m
default-interconnect                     ClusterIP   172.30.3.241     <none>        5672/TCP,8672/TCP,55671/TCP,5671/TCP,5673/TCP   93m
ibm-auditlogging-operator-metrics        ClusterIP   172.30.216.249   <none>        8383/TCP,8686/TCP                               11h
prometheus-operated                      ClusterIP   None             <none>        9090/TCP                                        93m
service-telemetry-operator-metrics       ClusterIP   172.30.11.66     <none>        8383/TCP,8686/TCP                               11h
smart-gateway-operator-metrics           ClusterIP   172.30.145.199   <none>        8383/TCP,8686/TCP                               11h
----

> . Note the port and service name to expose as a route to allow for external access, for example service `prometheus-operated` and port `9090`

Reviewer 2: ```suggestion
. Take note of the port and service name to expose as a route, for example, service `prometheus-operated` and port `9090`.
```

> . Expose the `prometheus-operated` service as an edge route and redirect insecure traffic to the secure endpoint of port `9090`

Reviewer 2: ```suggestion
. Expose the `prometheus-operated` service as an edge route and redirect insecure traffic to the secure endpoint of port `9090`:
```

+
[source,bash,options="nowrap",subs="+quotes"]
----
$ oc create route edge metrics-store --service=prometheus-operated --insecure-policy="Redirect" --port=9090
----
+
----
route.route.openshift.io/metrics-store created
----

> . To verify and find the exposed external DNS for the route use the `oc get route` command

Reviewer 2: ```suggestion
. To verify and find the exposed external DNS for the route, use the `oc get route` command:
```

+
[source,bash]
----
$ oc get route metrics-store -ogo-template='{{.spec.host}}'
----
+
----
metrics-store-service-telemetry.apps.infra.watch
----

> . The `prometheus-operated` service will now be avalable at the exposed DNS address, e.g. https://metrics-store-service-telemetry.apps.infra.watch

Reviewer 2: ```suggestion
+
The `prometheus-operated` service is now available at the exposed DNS address, for example, https://metrics-store-service-telemetry.apps.infra.watch
```
